File: dkr/2/dkr_2.pas
function IsMonotone(Arr: array of integer): boolean;
begin
  var pairs := Arr.Pairwise();
  var decreasing := pairs.All(p -> p[0] >= p[1]);
  var increasing := pairs.All(p -> p[0] <= p[1]);
  
  Result := decreasing or increasing
end;


procedure TaskOne;
begin
  var n := ReadInteger('Размер массива: ');
  Print('Элементы массива: ');
  var Arr := ReadArrInteger(n);
  
  Println(IsMonotone(Arr[::2]));
end;


procedure TaskTwo;
begin
  var Str := ReadString('Введите строку');
  var StrSet: set of string := [];
  var counter := 0;
  
  for var i := 1 to Str.Length do
  begin
    if i mod 3 <> 0 then continue;
    if Str[i-2:i+1] in StrSet then
      continue
    else
    begin
      StrSet.Add(Str[i-2:i+1]);
      Inc(counter);
    end
  end;
  Println(counter);
end;

File: dkr/2/dkr_2.py
def is_monotone(lst: list[int]) -> bool:
    pairs = list((lst[i], lst[i+1]) for i in range(len(lst)-1))
    increasing = all(p[0] <= p[1] for p in pairs)
    decreasing = all(p[0] >= p[1] for p in pairs)
    return decreasing or increasing


def task_one():
    print('Элементы списка: ', end='')
    lst = list(map(int, input().split()))
    print(is_monotone(lst[::2]))


def task_two():
    string = input('Введите строку: ')
    str_set = set()
    counter = 0

    for i in range(1, len(string)+1):
        if i % 3 != 0:
            continue
        substring = string[i-3:i]
        if substring in str_set:
            continue
        else:
            str_set.add(string[i-3:i])
            counter += 1
    print(counter)